patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.TML.cs

```diff
@@ -74,12 +74,19 @@ internal static bool TryGetModDownloadItem(string modSlug, out ModDownloadItem i
 	{
 		item = null;
 
-		var query = new QueryHelper.AQueryInstance(new QueryParameters() { searchModSlugs = new string[] { modSlug } });
-		if (!query.TrySearchByInternalName(out var items))
+		var query = new QueryHelper.AQueryInstance(new QueryParameters() { queryType = QueryType.SearchDirect });
+		if (!query.TrySearchByInternalName(modSlug, out item))
 			return false;
 
-		item = items[0];
-		return item != null; // TODO, return value is ambiguous between a connection error and the mod not existing on workshop, currently both are logged as an error and the item is skipped
+		if (item == null) {
+			// search of all mods that were published by this user - Does not work with co-published/non-owner
+			// this should catch private / friends-only visibility
+			query.queryParameters.queryType = QueryType.SearchUserPublishedOnly;
+			if (!query.TrySearchByInternalName(modSlug, out item))
+				return false;
+		}
+
+		return true;
```

Comment:
`TrySearchByInternalName` only returns `false` when the query fails. Previously, `TryGetModDownloadItem` would return `false` if `item == null`. Now it returns `true`. Perhaps the final line should still be `return item != null`?

In general, I think this approach of using `TryGet` with `false` indicating error, and requiring users to do an additional null check is quite error prone. In previous cleanups we've tried to keep nulls out of the system, and use exceptions or more detailed error codes.

It's okay to return `false` on both error, and item not existing, since this just classifies the item not existing as an error.

Reply:
In this case, the only prior usage of `TryGetModDownloadItem` is this missing deps downloader in `Interface.cs`. It needs a `null` check now.

If we had nullable assertions enabled on the codebase, this would be more okay. Unfortunately, an `Option` type in the system C# libraries is only planned for a future version of the language, but it might be worth using one of them to explicitly encode these presence or absences in the type system. Perhaps `Optional<T>` from `Microsoft.CodeAnalysis` which we already have available. (`null` is replaced by `default(Optional<T>)`)

I note that `TryGroupSearchByInternalName` also has to handle these `null` items, and `TryGetPublishIdByInternalName` converts them to `"0"` entries.

The two functions which call `TryGetPublishIdByInternalName` both end up discarding these `"0"` entries anyway, but perhaps there are times when `ModDownloadItem.PublishId.m_ModPubId` could also be `"0"`?

Reply:
| I think this approach of using TryGet with false indicating error is error-prone
Suggesting separate PR to clean up the code around the null/false. 
Proposing that if 'Steam API unavailable', that it throw an exception instead of return false. 
Change 'false' to actually indicate the 'item not found' . 

| perhaps there are times when ModDownloadItem.PublishId.m_ModPubId could also be "0"
0 is the default for type. The actual steamworks api will return a zero in several methods that fail.